pyuvdata/uvdata/uvh5.py

```diff
@@ -312,12 +321,56 @@ def __init__(
         self.__blts_are_rectangular = blts_are_rectangular
         self.__time_first = time_axis_faster_than_bls
         self.__blt_order = blt_order
+        self._recompute_nbls = recompute_nbls
 
     def __del__(self):
         """Close the file when the object is deleted."""
         if self.__file:
             self.__file.close()
 
+    def __getstate__(self):
+        """Get the state of the object."""
+        state = {
+            k: v
+            for k, v in self.__dict__.items()
+            if k
+            not in (
+                "_FastUVH5Meta__file",
+                "_FastUVH5Meta__header",
+                "_FastUVH5Meta__datagrp",
+                "header",
+                "datagrp",
+            )
+        }
+        return state
+
+    def __setstate__(self, state):
+        """Set the state of the object."""
+        self.__dict__.update(state)
+        self.__file = None
+        self.open()
```

Comment:
can you explain the purpose of this to me?

Reply:
So the `__getstate__` / `__setstate__` are run before and after pickling. The `self.__file` attribute is not pickleable, so in `__getstate__` I remove it, and in `__setstate__` I restore it from the path. Having the class be pickleable is really useful for doing things like deepcopy and multiprocessing.